src/dft/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+endif(COMPILER_SUPPORTS_ADVSIMD)
+
+# Target mkunroll
+
+set(TARGET_MKUNROLL "mkunroll_exe")
+add_executable(${TARGET_MKUNROLL} EXCLUDE_FROM_ALL mkunroll.c)
+set_target_properties(${TARGET_MKUNROLL} PROPERTIES ${COMMON_TARGET_PROPERTIES})
+
+# Target mkdispatch
+
+set(TARGET_MKDISPATCH "mkdispatch_exe")
+add_executable(${TARGET_MKDISPATCH} EXCLUDE_FROM_ALL mkdispatch.c)
+set_target_properties(${TARGET_MKDISPATCH} PROPERTIES ${COMMON_TARGET_PROPERTIES})
+
+# Target dispatchparam.h
+
+add_custom_command(OUTPUT dispatchparam.h
+  COMMENT "Generating dispatchparam.h"
+  COMMAND ${TARGET_MKDISPATCH} paramonly ${MAXBUTWIDTH} ${ISALIST_DP} > ${CMAKE_CURRENT_BINARY_DIR}/dispatchparam.h
+  DEPENDS ${TARGET_MKDISPATCH}
+  )
+add_custom_target(dispatchparam.h_generated SOURCES ${CMAKE_CURRENT_BINARY_DIR}/dispatchparam.h)
+
+# Target dispatch*.h
+
+foreach(T ${LIST_SUPPORTED_FPTYPE})
+  list(GET LISTSHORTTYPENAME ${T} ST)                       # ST is "dp", for example
+  string(TOUPPER ${ST} CST)                                 # CST is "DP"
+  list(GET LISTLONGTYPENAME ${T} LT)                        # LT is "double"
+  list(GET LISTTYPEID ${T} ID)                              # ID is 1
+  
+  string(CONCAT S "dispatch" ${ST} ".h")                    # S is dispatchdp.h
+  add_custom_command(OUTPUT ${S}
+    COMMENT "Generating ${S}"
+    COMMAND ${TARGET_MKDISPATCH} ${LT} ${MAXBUTWIDTH} ${ISALIST_${CST}} > ${S}
+    DEPENDS ${TARGET_MKDISPATCH}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    )
+
+  string(CONCAT G ${S} "_generated")                        # G is dispatchdp.h_generated
+  add_custom_target(${G} SOURCES ${S})
+endforeach()
+
+# Target dftcommon.o
+
+add_library(dftcommon_obj OBJECT dftcommon.c dftcommon.h ${CMAKE_CURRENT_BINARY_DIR}/dispatchparam.h ${CMAKE_BINARY_DIR}/include/sleef.h)
+add_dependencies(dftcommon_obj ${TARGET_HEADERS} dispatchparam.h_generated)
+set_source_files_properties(${CMAKE_BINARY_DIR}/include/sleef.h PROPERTIES GENERATED TRUE)
+set_target_properties(dftcommon_obj PROPERTIES ${COMMON_TARGET_PROPERTIES})
+target_compile_definitions(dftcommon_obj PRIVATE ${COMMON_TARGET_DEFINITIONS})
+
+# Target dft*.o
+
+foreach(T ${LIST_SUPPORTED_FPTYPE})
+  list(GET LISTSHORTTYPENAME ${T} ST)                       # ST is "dp", for example
+  
+  string(CONCAT G "dft" ${ST} "_obj")                       # G is "dftdp_obj"
+  string(CONCAT S "dispatch" ${ST} ".h")                    # S is "dispatchdp.h"
+  add_library(${G} OBJECT dft.c dftcommon.h ${S})
+  string(CONCAT SG ${S} "_generated")                       # SG is "dispatchdp.h_generated"
+  add_dependencies(${G} ${SG})
+  set_target_properties(${G} PROPERTIES ${COMMON_TARGET_PROPERTIES})
+  list(GET LISTTYPEID ${T} ID)                              # ID is 1
+  target_compile_definitions(${G} PRIVATE BASETYPEID=${ID} ${COMMON_TARGET_DEFINITIONS})
+endforeach()
+
+# Copy unroll0.org to ${CMAKE_CURRENT_BINARY_DIR}
+
+add_custom_target(unroll0.org_source
```

Comment:
Better `add_custom_command` instead so that you can specify proper dependencies between input and output files

Reply:
This is where I am having trouble.
I thought that dependency between source ${CMAKE_CURRENT_SOURCE_DIR}/unroll0.org and target unroll0.org_source can be specified clearly with this.
I will check if add_custom_target works better tomorrow since I am away on business trip today.

Reply:
Note that SOURCES in add_custom_target doesn't express a dependency, but just notifies that the files should be visible to the IDE project. A custom target is always considered out-of-date and always built. That's why it is better to rely on `add_custom_command` (but you can then add a dependency between a custom target and a custom command)